docs/framework/wpf/advanced/hosting-win32-content-in-wpf.md

```diff
@@ -1,357 +1,368 @@
+
+If your [!INCLUDE[TLA2#tla_win32](../../../../includes/tla2sharptla-win32-md.md)] logic is already packaged as a control, then your `BuildWindowCore` implementation is little more than a call to `CreateWindow`. For example, to create a [!INCLUDE[TLA2#tla_win32](../../../../includes/tla2sharptla-win32-md.md)] LISTBOX control in [!INCLUDE[TLA#tla_cpp](../../../../includes/tlasharptla-cpp-md.md)]:
+
+```cpp
+virtual HandleRef BuildWindowCore(HandleRef hwndParent) override {
+    HWND handle = CreateWindowEx(0, L"LISTBOX",
+    L"this is a Win32 listbox",
+    WS_CHILD | WS_VISIBLE | LBS_NOTIFY
+    | WS_VSCROLL | WS_BORDER,
+    0, 0, // x, y
+    30, 70, // height, width
+    (HWND) hwndParent.Handle.ToPointer(), // parent hwnd
+    0, // hmenu
+    0, // hinstance
+    0); // lparam
+
+    return HandleRef(this, IntPtr(handle));
+}
+
+virtual void DestroyWindowCore(HandleRef hwnd) override {
+    // HwndHost will dispose the hwnd for us
+}
+```
+
+But suppose the [!INCLUDE[TLA2#tla_win32](../../../../includes/tla2sharptla-win32-md.md)] code is not quite so self-contained? If so, you can create a [!INCLUDE[TLA2#tla_win32](../../../../includes/tla2sharptla-win32-md.md)] dialog box and embed its contents into a larger [!INCLUDE[TLA2#tla_winclient](../../../../includes/tla2sharptla-winclient-md.md)] application. The sample shows this in [!INCLUDE[TLA#tla_visualstu](../../../../includes/tlasharptla-visualstu-md.md)] and [!INCLUDE[TLA#tla_cpp](../../../../includes/tlasharptla-cpp-md.md)], although it is also possible to do this in a different language or at the command line.
+
+Start with a simple dialog, which is compiled into a [!INCLUDE[TLA#tla_cpp](../../../../includes/tlasharptla-cpp-md.md)] [!INCLUDE[TLA2#tla_dll](../../../../includes/tla2sharptla-dll-md.md)] project.
+
+Next, introduce the dialog into the larger [!INCLUDE[TLA2#tla_winclient](../../../../includes/tla2sharptla-winclient-md.md)] application:
+
+- Compile the [!INCLUDE[TLA2#tla_dll](../../../../includes/tla2sharptla-dll-md.md)] as managed (`/clr`)
+
+- Turn the dialog into a control
+
+- Define the derived class of <xref:System.Windows.Interop.HwndHost> with `BuildWindowCore` and `DestroyWindowCore` methods
+
+- Override `TranslateAccelerator` method to handle dialog keys
+
+- Override `TabInto` method to support tabbing
+
+- Override `OnMnemonic` method to support mnemonics
+
+- Instantiate the <xref:System.Windows.Interop.HwndHost> subclass and put it under the right [!INCLUDE[TLA2#tla_winclient](../../../../includes/tla2sharptla-winclient-md.md)] element
+
+### Turn the Dialog into a Control
+
+You can turn a dialog box into a child HWND using the WS_CHILD and DS_CONTROL styles. Go into the resource file (.rc) where the dialog is defined, and find the beginning of the definition of the dialog:
+
+```rc
```

Comment:
I don't think rc is supported natively. Should we just leave this one empty?

Reply:
Yeah, it can be removed, I just added it since it matched the file type and figured it avoids a MD040 warning later

Reply:
```suggestion
```
```